This notebook analyzes the northernmost and easternmost airports in the United States, then uses the data to find the windiest weather of New York area airports on Feb 12, 2013. The data used comes from the nycflights13 dataset, specifically the airports.csv and weather.csv files.

In [16]:
import pandas as pd

# Load datasets from local files
airports = pd.read_csv("airports.csv")
weather = pd.read_csv("weather.csv")

# Ensure only U.S. airports are considered (assuming 'country' column exists)
if 'country' in airports.columns:
    us_airports = airports[airports['country'] == 'US']  # Filter for U.S. airports
else:
    us_airports = airports.copy()  # If no country column, assume all airports are U.S.

# Filter out small airports, considering only large ones (if a column exists for classification)
if 'alt' in us_airports.columns:  # Some datasets include altitude; if not, assume all airports are relevant
    major_airports = us_airports[us_airports['alt'] > 0]  # Example filter; adjust based on dataset structure
else:
    major_airports = us_airports.copy()

# Top 10 Northernmost Airports in the U.S.
top_northernmost_airports = major_airports[['faa', 'name', 'lat']].sort_values(by='lat', ascending=False).head(10)
print("\nTop 10 Northernmost Airports in the U.S.:")
print(top_northernmost_airports)

# Find the easternmost airport and the top 10 easternmost airports
easternmost_airport = major_airports.loc[major_airports['lon'].idxmax()]
top_easternmost_airports = major_airports[['faa', 'name', 'lat', 'lon']].sort_values(by='lon', ascending=False)

# Top 10 highest latitude among the easternmost airports
top_easternmost_by_latitude = top_easternmost_airports.head(10).sort_values(by='lat', ascending=False)

print("\nEasternmost Airport in the U.S.:")
print(easternmost_airport[['faa', 'name', 'lat', 'lon']])

print("\nTop 10 Easternmost Airports in the U.S.:")
print(top_easternmost_airports[['faa', 'name', 'lat', 'lon']])

print("\nTop 10 Highest Latitude Easternmost Airports in the U.S.:")
print(top_easternmost_by_latitude[['faa', 'name', 'lat', 'lon']])

# Ensure time_hour column is in datetime format for weather data
weather['time_hour'] = pd.to_datetime(weather['time_hour'], errors='coerce')

# Filter weather data for February 12, 2013, in New York area airports (Only JFK and LGA)
ny_airports = ['JFK', 'LGA']
weather_ny_feb12 = weather[(weather['origin'].isin(ny_airports)) & (weather['time_hour'].dt.date == pd.to_datetime('2013-02-12').date())]

# Find the airport with the highest wind speed on this date
if not weather_ny_feb12.empty:
    windiest_airport = weather_ny_feb12.loc[weather_ny_feb12['wind_speed'].idxmax()]
    print("\nWindiest New York Area Airport on February 12, 2013:")
    print(windiest_airport[['origin', 'wind_speed']])
else:
    print("\nNo weather data available for the specified date and airports.")


Top 10 Northernmost Airports in the U.S.:
      faa                                        name        lat
417   EEN                     Dillant Hopkins Airport  72.270833
230   BRW                  Wiley Post Will Rogers Mem  71.285446
110   AIN                          Wainwright Airport  70.638056
708   K03                               Wainwright As  70.613378
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300
1363  UUK                        Ugnu-Kuparuk Airport  70.330833
982   NUI                             Nuiqsut Airport  70.210000
1197  SCC                                   Deadhorse  70.194750
232   BTI                          Barter Island Lrrs  70.133989
1084  PIZ                              Point Lay Lrrs  69.732875

Easternmost Airport in the U.S.:
faa              SYA
name    Eareckson As
lat        52.712275
lon        174.11362
Name: 1290, dtype: object

Top 10 Easternmost Airports in the U.S.:
      faa                           name        lat     

The northernmost airport is Dillant Hopkins Airport (FAA Code: EEN), located at latitude 72.27.

The easternmost airport is Eareckson As (FAA Code: SYA), located at longitude 174.11.

The windiest airport in the New York area was LaGuardia (LGA) with a wind speed of 23.02 m/s.

In conclusion, adding the top 10 northernmost airports in the United States provided a clearer picture of the actual northernmost airport, as previous research suggested otherwise. Additionally, existing research did not identify Eareckson Air Station as the easternmost airport in the U.S., likely because it is no longer an active civilian airport and is primarily used for refueling and emergency diversions. Initially, my research and dataset indicated that Newark Airport was the windiest airport on February 12, 2013. However, since Newark is not located in New York, I modified the code to ensure that only New York airports were considered.